File: src/javascript/NumberOfStringsThatAppearAsSubstringsInWord.js
const numOfStrings = function (patterns, word) {
    let count = 0;
    for (let i = 0; i < patterns.length; i++) {
        if (word.includes(patterns[i])) {
            count++;
        }
    }
    return count;
};

const data = ["a", "abc", "bc", "d"];
let word = "abc";

let number = numOfStrings(data, word);
console.log(number);

const numOfStringsOneLine = ((patterns, word) => patterns.filter(x => word.includes(x)).length);

let result = numOfStringsOneLine(data, word);
console.log(result);

File: src/NumberOfStringsThatAppearAsSubstringsInWord.java
public class NumberOfStringsThatAppearAsSubstringsInWord {
    public static void main(String[] args) {
        String[] patterns = {"a", "abc", "bc", "d"};
        var word = "abc";
        int i = numOfStrings(patterns, word);
        System.out.println(i);
    }

    public static int numOfStrings(String[] patterns, String word) {
        int count = 0;
        for (String pattern : patterns) {
            if (word.contains(pattern)) {
                count++;
            }
        }
        return count;
    }
}
